### PR TITLE
Removes ability to put collars on jumpsuits

### PR DESCRIPTION
**Pull request by AffectedArc07**
# Preliminary Note
This has already been discussed internally between admins and heads, and we have decided we want this. Please do not reeeee in the comments.

## What Does This PR Do
Removes the ability to put collars on jumpsuits.

## Why It's Good For The Game
People only ever add collars to jumpsuits for weird petplay fetish things, which we dont want here. Its time to stop this.

## Changelog
:cl: AffectedArc07
del: You can no longer put pet collars on jumpsuits
/:cl:

### CODE REVIEW

**Comment by DoctorDrugs** [comment]
REEEEEEEEEEEEEEEEEEEEEEEE (because you said not to and i quote "i will repo ban you")

**Comment by Necaladun** [comment]
REEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEally about time.

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -666,6 +666,11 @@ BLIND     // can't see anything
 		3 = Report location
 		*/
 	var/list/accessories = list()
+
+	/// List of blacklisted accessory types
+	var/list/blacklisted_accessory_typecache = list(
```
Comment (dearmochi) [inline code/modules/clothing/clothing.dm:671]: `typecacheof`?

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -685,11 +690,17 @@ BLIND     // can't see anything
 	else
 		return FALSE
 
+	if(A.type in blacklisted_accessory_typecache)
```
Comment (dearmochi) [inline code/modules/clothing/clothing.dm:693]: `is_type_in_typecache`?

**Comment by farie82** [comment]
Why keep it as a `obj/item/clothing/accessory` if you can't equip it as an accessory? Sounds like this just needs to become something else that is only useable by pets or so

**Comment by Aurorablade** [comment]
Look if people are using it for pet play fetish things we need to crack down on them or add some kind of alternative like actually military dogtags. Editable in the same vein if we are going to remove this. I really dislike removing based on the people using it for weird fetish crap.

**Comment by AffectedArc07** [comment]
> Why keep it as a `obj/item/clothing/accessory` if you can't equip it as an accessory? Sounds like this just needs to become something else that is only useable by pets or so

Because refactoring the pathtypes is out the scope of this PR, and I dont want to inadvertently break corgi access by doing that

**Comment by Spacemanspark** [comment]
@aurorablade
I'd be just fine with adding an alternative that wouldn't be used for kinky purposes so easily. Military dogtags would be neat. Someone has also suggested chokers, which I believe is far less of a fetish item and would do well for those looking for the 'punk kid' angle. 

That's all beyond the scope of this PR however.

This item has long been a problem. It can still be used as a pet accessory as it's primary intended use.

**Comment by AffectedArc07** [comment]
Tagged `Controversial` even though it really shouldn't be

**Comment by Cocacolagua** [comment]
lol

**Comment by ZanderDoggo** [comment]
I believe that there should be an alternative, or just adding in a new rule about it for using the collars in an ERP way. The main problem I have with removing collars is that you then can't clip IDs to them, that's what I use them for. So having dog tags as an alternative where you can still clip an ID to it would be good.

**Comment by Spacemanspark** [comment]
>just adding in a new rule about it for using the collars in an ERP way

Our server rules already cover this sort of behavior.
The problem is that people, even regulars, still toe the line with the item... and honestly there's no real good reason for it to exist as it is. We can tell people not to do something over and over, but when the behavior is consistent across several different types of people, we can't really just take the usual ban hammer approach. 

Again, alternatives would be great to have in a future PR should someone want to sprite up new things. But as it is, the item treads far too close to the line... and we've only really ever seen one type of person using it for anything but a gross fetish purpose or joke.

**Comment by procdrone** [comment]
what the fuck

**Comment by farie82** [comment]
> > Why keep it as a `obj/item/clothing/accessory` if you can't equip it as an accessory? Sounds like this just needs to become something else that is only useable by pets or so
> 
> Because refactoring the pathtypes is out the scope of this PR, and I dont want to inadvertently break corgi access by doing that

Yet will leave technical debt this way. An accessory that can't be used as one. Why is it one then?

**Comment by AffectedArc07** [comment]
> > > Why keep it as a `obj/item/clothing/accessory` if you can't equip it as an accessory? Sounds like this just needs to become something else that is only useable by pets or so
> > 
> > 
> > Because refactoring the pathtypes is out the scope of this PR, and I dont want to inadvertently break corgi access by doing that
> 
> Yet will leave technical debt this way. An accessory that can't be used as one. Why is it one then?

If you wanna clean it up after the fact and double check things, you are more than welcome to

**Comment by Carthusia** [comment]
Wait...this is actually happening? Seriously?

Is it in like the darkest depths of maint that no light shall escape from, and all eyes who behold it shall be clawed out from thine head as to prevent madness?

**Comment by Spacemanspark** [comment]
No, no, the madness has been amongst you this whole time, in fact. Walking among you as you stroll to the bridge to gaze upon the captain, watching you.

If you are not careful, the curse will catch you too, dear Kar-Ski. Beware.

**Comment by Carthusia** [comment]
The mad laughs of dead gods...

**Review by Kyep — APPROVED** [review]

**Comment by Kyep** [comment]
For clarity: I wasn't the first head to vote in favor of this, so my approval isn't necessary for this to pass, but I'll still add my approval anyway because I feel its important for our community to know that this change has broad support among the staff team.

**Review by variableundefined — APPROVED** [review]

**Comment by Bkulli-VI** [comment]
But where will I hide the spare now?

**Pull request merged**